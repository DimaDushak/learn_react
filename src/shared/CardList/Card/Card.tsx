import React from 'react';
import type { PostData } from '../../../utils/hooks';
import postContext from '../../context/postContext';
import styles from './card.css';
import CardDropdown from './CardDropdown';
import ControlBlock from './ControlBlock';
import Preview from './Preview';
import TextBlock from './TextBlock';

interface Props {
    post: PostData['data'];
}

export default function Card({ post }: Props): JSX.Element {
    return (
        <li className={styles.card}>
            <postContext.Provider value={post}>
                <TextBlock />
                <Preview />
                <ControlBlock />
                <CardDropdown />
            </postContext.Provider>
        </li>
    );
}
