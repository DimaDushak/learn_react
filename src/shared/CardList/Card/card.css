.card {
    background-color: var(--white);
    display: flex;
    flex-wrap: wrap;
    margin: 10px 0;
    position: relative;
}

.card > div {
    flex-basis: 100%;
}

.card:hover {
    outline: 1px solid var(--greyD9);
}

.card:hover h2 a {
    color: var(--orange);
}

@media (min-width: 1024px) {
    .card {
        align-items: center;
        flex-wrap: nowrap;
        margin: 0;
        padding: 20px 40px 20px 37px;
    }

    .card > div {
        flex-basis: auto;
    }
}

@media (min-width: 1540px) {
    .card {
        padding: 20px 45px 20px 72px;
    }
}
