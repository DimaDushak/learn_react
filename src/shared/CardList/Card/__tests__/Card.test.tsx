import { render } from '@testing-library/react';
import React from 'react';
import Card from '../Card';

test('should render Card (snapshot)', () => {
    const { container } = render(<Card />);

    expect(container.firstChild).toMatchSnapshot();
});
