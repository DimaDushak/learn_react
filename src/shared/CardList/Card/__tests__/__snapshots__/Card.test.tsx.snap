// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Card (snapshot) 1`] = `
<li
  class="card"
>
  <div
    class="textBlock"
  >
    <div
      class="authorInfoBlock"
    >
      <a
        class="userBlock userBlockSizes"
        href="#user"
      >
        <img
          alt="#"
          class="authorAvatar"
          src="https://cdn.dribbble.com/users/2202649/avatars/small/2044ffe8e8848aeb80e514229c21c515.png?1616573826"
        />
        <span
          class=""
        >
          Константин Кодов
        </span>
      </a>
      <span
        class="publishTime"
      >
        <span
          class="publish"
        >
          опубликовано
        </span>
        8 часов назад
      </span>
    </div>
    <h2
      class="postHeading"
    >
      <a
        class="postLink"
        href="#post"
      >
        Следует отметить, что новая модель организационной деятельности и деятельности и деятельности и деятельности
      </a>
    </h2>
  </div>
  <div
    class="preview"
  >
    <img
      alt="#"
      class="previewImage"
      src="/static/image/Rectangle.png"
    />
  </div>
  <div
    class="controlBlock"
  >
    <div>
      <button
        class="upButton"
        type="button"
      >
        <svg
          fill="none"
          height="10"
          viewBox="0 0 19 10"
          width="19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.5 0L0 10H19L9.5 0Z"
            fill="#D9D9D9"
          />
        </svg>
      </button>
      <span
        class="voteCount"
      >
        101
      </span>
      <button
        class="downButton"
        type="button"
      >
        <svg
          fill="none"
          height="10"
          viewBox="0 0 19 10"
          width="19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.5 0L0 10H19L9.5 0Z"
            fill="#D9D9D9"
          />
        </svg>
      </button>
    </div>
    <div>
      <button
        class="messageButton"
        type="button"
      >
        <svg
          fill="none"
          height="14.17"
          viewBox="0 0 14.17 14.17"
          width="14.17"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.75 0H1.41667C0.6375 0 0 0.6375 0 1.41667V9.91667C0 10.6958 0.6375 11.3333 1.41667 11.3333H11.3333L14.1667 14.1667V1.41667C14.1667 0.6375 13.5292 0 12.75 0ZM11.3333 8.5H2.83333V7.08333H11.3333V8.5ZM11.3333 6.375H2.83333V4.95833H11.3333V6.375ZM11.3333 4.25H2.83333V2.83333H11.3333V4.25Z"
            fill="#999999"
          />
        </svg>
      </button>
      <span>
        14
      </span>
    </div>
    <div>
      <button
        class="roundButton"
        type="button"
      >
        <svg
          fill="none"
          height="8"
          viewBox="0 0 13 8"
          width="13"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.1 0L10.5885 1.52667L7.4165 4.78L4.8165 2.11333L0 7.06L0.9165 8L4.8165 4L7.4165 6.66667L11.5115 2.47333L13 4V0H9.1Z"
            fill="#999999"
          />
        </svg>
      </button>
      <button
        class="roundButton"
        type="button"
      >
        <svg
          class="shareIcon"
          fill="none"
          height="10"
          viewBox="0 0 8 10"
          width="8"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 9.89558C9.49333 9.89558 9.04 10.1064 8.69333 10.4367L3.94 7.52008C3.97333 7.35843 4 7.19679 4 7.02811C4 6.85944 3.97333 6.69779 3.94 6.53614L8.64 3.64759C9 3.999 9.47333 4.21687 10 4.21687C11.1067 4.21687 12 3.2751 12 2.10843C12 0.941767 11.1067 0 10 0C8.89333 0 8 0.941767 8 2.10843C8 2.27711 8.02667 2.43875 8.06 2.6004L3.36 5.48896C3 5.13755 2.52667 4.91968 2 4.91968C0.893333 4.91968 0 5.86145 0 7.02811C0 8.19478 0.893333 9.13655 2 9.13655C2.52667 9.13655 3 8.91867 3.36 8.56727L8.10667 11.491C8.07333 11.6386 8.05333 11.7932 8.05333 11.9478C8.05333 13.0793 8.92667 14 10 14C11.0733 14 11.9467 13.0793 11.9467 11.9478C11.9467 10.8163 11.0733 9.89558 10 9.89558Z"
            fill="#999999"
          />
        </svg>
      </button>
      <button
        class="roundButton"
        type="button"
      >
        <svg
          fill="none"
          height="10"
          viewBox="0 0 8 10"
          width="8"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0.571429 8.88889C0.571429 9.5 1.08571 10 1.71429 10H6.28571C6.91429 10 7.42857 9.5 7.42857 8.88889V2.22222H0.571429V8.88889ZM1.71429 3.33333H6.28571V8.88889H1.71429V3.33333ZM6 0.555556L5.42857 0H2.57143L2 0.555556H0V1.66667H8V0.555556H6Z"
            fill="#999999"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <div>
        <button
          class="menuButton"
          type="button"
        >
          <svg
            fill="none"
            height="5"
            viewBox="0 0 20 5"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="17.5"
              cy="2.5"
              fill="#D9D9D9"
              r="2.5"
              transform="rotate(90 17.5 2.5)"
            />
            <circle
              cx="10"
              cy="2.5"
              fill="#D9D9D9"
              r="2.5"
              transform="rotate(90 10 2.5)"
            />
            <circle
              cx="2.5"
              cy="2.5"
              fill="#D9D9D9"
              r="2.5"
              transform="rotate(90 2.5 2.5)"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</li>
`;
